lib/ace/ext/papeeria/highlighter.coffee

```diff
@@ -76,6 +77,20 @@ define((require, exports, module) ->
                     else
                         rightNearest = rightCandidate
             key++
+
+        #Special case related to positionLeftwards has not next char and positionLeftWards.column += 1 does nothing.
+        if session.getLine(positionRightwards.row + 1) == '' 
```

Comment:
This can easily be the case when you just have an empty line after the cursor, and tons of text afterwards.